components/style/viewport.rs

```diff
@@ -92,6 +93,50 @@ declare_viewport_descriptor! {
     Orientation(Orientation),
 }
 
+impl ToCss for ViewportDescriptor {
```

Comment:
Yes, please extend the macro. It will need to take in a string as well, so the invocation will be something like

```
 declare_viewport_descriptor! {
     "min-width": MinWidth(ViewportLength),
     "max-width": MaxWidth(ViewportLength),
...
```

You could also do 

```
 declare_viewport_descriptor! {
     min-width: MinWidth(ViewportLength),
     max-width: MaxWidth(ViewportLength),
...
```

and use `stringify!()` on the identifiers. It may be easy to make the macro follow sets work that way. If you're having trouble, I can fix this up for you, so don't worry too much about it.